### PR TITLE
refactor: restructure repo as a single monorepo

### PR DESCRIPTION
**Pull request by andrewthauer**
This is a large restructuring to improve the local development experience of the techdocs-cli. 

#### Why

I encountered a frustrating time trying to test #88 and other PR contributions. In particular when it comes to the embedded-techdocs project. The [README](docs/README.md) talks about some issues with the project setup due to the fact it is a has a monorepo in another monorepo. This is an attempt to make the experience working with this repo much more pleasant and working towards more consistency with the Backstage core repo experience.

#### What

- Remove the `embeded-techdocs` project (monorepo). This moves `packages/embedded-techdocs/packages/app` to `packages/embedded-techdocs-app` and removes the need for the nested lerna monorepo.
- Updated the npm scripts to use more standard scripts, leveraging the backstage-cli. The one exception is the `cli` which needs to copy the bundle of the embedded app into it's dist directory.
- Fixed up some eslint issue (based on the backstage/cli config), etc. across various files
- Fixed up most `tsc` issues
- Removed the `dist-types` and gitignored it since they should not be required anymore
- Finally made the `techdocs-cli` support using served version of the `embedded-techdocs-app` so you can test changes without having to rebuild each time.

New structure based on these changes:

```
.
├── docs
│   └── assets
└── packages
    ├── embedded-techdocs-app
    └── techdocs-cli
```

#### Next Steps

- Update techdocs-cli to use prettier settings from backstage/cli config and fix those files
- Fix up some additional eslint rules overriden and enable in CI
- Add additional unit tests

### CODE REVIEW

File: packages/techdocs-cli/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  extends: [require.resolve('@backstage/cli/config/eslint')],
+  overrides: [
+    {
+      files: ['**/*.ts?(x)'],
+      rules: {
+        'no-restricted-imports': 0,
```
Comment (andrewthauer) [inline packages/techdocs-cli/.eslintrc.js:7]: This was being flagged for things like `import path from 'path`` etc. Just trying to keep the number of changes to a minimum.

File: packages/techdocs-cli/src/commands/generate/generate.ts [hunk 2]
```diff
@@ -85,6 +86,7 @@ export default async function generate(cmd: Command) {
         }
       : {}),
     logger,
+    ...(process.env.LOG_LEVEL === "debug" ? { logStream: stdout } : {}),
```
Comment (andrewthauer) [inline packages/techdocs-cli/src/commands/generate/generate.ts:89]: Sometimes errors from mkdocs are not output so this allows us to see the underlying issue.

File: packages/techdocs-cli/src/e2e.test.ts [hunk 3]
```diff
@@ -9,28 +9,28 @@ describe("end-to-end", () => {
...
-      }
-    );
+    jest.setTimeout(10000);
+    const proc = await executeTechDocsCliCommand(["generate", "--no-docker"], {
+      cwd: "../../",
+      killAfter: 8000,
+    });
+
+    // eslint-disable-next-line no-console
+    console.log(proc.combinedStdOutErr);
```
Comment (andrewthauer) [inline packages/techdocs-cli/src/e2e.test.ts:19]: Have been having problems with what seems like flaky tests. They often work locally, but fail in CI for me. This was an attempt to see the output when they do fail. Coincidentally, adding this made the tests pass (at least the first time). These may need to be re-worked in general.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+yarn build
+```
+
+You can now run:
+
+```sh
+yarn cli
+
+# or using the techdoc-cli binary directly (or as an alias)
+packages/techdocs-cli-bin/techdocs-cli
```
[the remark this answers is not in the document]
Comment (iamEAP) [reply README.md:26]: Although actually...  Perhaps we should merge this file with `docs/README.md` which currently serve the dual purpose of being a convenient TechDocs site to test with, but also the repo's current displayed `README`.

File: packages/techdocs-cli/src/e2e.test.ts (hunk 3 above)
Comment (iamEAP) [reply packages/techdocs-cli/src/e2e.test.ts:19]: An alternative suggestion: if you re-order the expectations below, you'll get `proc.combinedStdOutErr` printed as part of the test fail, while green builds don't get the console log spam.

**Review by iamEAP — COMMENTED** [review]
Thanks @andrewthauer!  This is _miles_ ahead of where we were.

I have a couple of small suggestions, but this otherwise looks good to merge from my POV.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (andrewthauer) [reply README.md:26]: Yeah, I didn't realize that GH will default to `docs/README.md` if the root readme doesn't exist. Makes sense to me. However, I've sometimes have been splitting up docs for consumers of the tools/lib/api/etc in `docs` and then using the root `README.md` for contributor notes/instructions. I'm indifferent and will follow your lead on what you think is best 😄.

[the remark this answers is not in the document]
Comment (iamEAP) [reply README.md:26]: That makes sense too. Perhaps then, taking the opportunity to strip out the user docs from the docs subfolder read me (including the build badge stuff)?

**Review by iamEAP — APPROVED** [review]
🔥 🔥 🔥  Many thanks, @andrewthauer.

**Pull request merged**